File: Data Structures & Algorithms/largest-unique-number/submission-2.cpp
class Solution {
   public:
    int largestUniqueNumber(vector<int>& nums) {
        unordered_map<int, int> counts;

        for (int num : nums) {
            counts[num] += 1;
        }

        int maxNumber = -1;

        for (const auto& [key, value] : counts) {
            if (value == 1) maxNumber = max(maxNumber, key);
        }

        return maxNumber;
    }
};

File: Data Structures & Algorithms/max-water-container/submission-2.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        max_area = 0;

        l, r = 0, len(heights) - 1;

        while l < r:
            h_l = heights[l]
            h_r = heights[r]

            max_area = max(max_area, (r - l) * min(h_r, h_l))

            if h_l < h_r:
                l += 1
            else:
                r -= 1

        return max_area;